File: flight-catalogue/flight-catalogue-model/src/main/java/com/pulse/air/flightcatalogue/contract/AircraftService.java
package com.pulse.air.flightcatalogue.contract;

import com.pulse.air.common.model.BaseSearchRequest;
import com.pulse.air.commons.contract.BaseCRUDService;
import com.pulse.air.flightcatalogue.model.aircraft.AircraftRequest;
import com.pulse.air.flightcatalogue.model.aircraft.AircraftResponse;

public interface AircraftService extends BaseCRUDService<AircraftResponse, AircraftRequest, BaseSearchRequest> {

}

File: airpulse-ui/src/app/pages/admin/services/aircraft.service.ts
import { HttpClient } from "@angular/common/http";
import { Injectable } from "@angular/core";
import { BaseService } from "../../shared/services/base.service";
import { LoaderService } from "../../shared/services/loader.service";
import { AircraftResponse, AircraftRequest } from "../model/aircraft.model";

@Injectable({
    providedIn: 'root',
})
export class AircraftService extends BaseService<AircraftResponse, AircraftRequest> {
    constructor(protected override http: HttpClient, protected override loader: LoaderService) {
        super(http, 'fc/aircraft', loader);
    }
}